### DISCUSSION TITLE
Get note value?

### DISCUSSION THREAD

**Discussion by cad12345** [melanchall/drywetmidi#213] [Q&A] [answered]
Hi,

How do you get the note values of all notes (such as whether they are quarter, half, whole notes etc)? I've seen info about it, but I can't work out how to do it. Thanks for your great work!

**Comment by melanchall** [comment, marked as answer]
Hi,

To get all notes:

```csharp
var notes = midiFile.GetNotes();
```

To get a note's length in terms of fraction of whole note:

```csharp
var length = note.LengthAs<MusicalTimeSpan>(tempoMap);
```

Then you can get numerator and denominator of the `length`.

To get tempo map:

```csharp
var tempoMap = midiFile.GetTempoMap();
```

**Reply by cad12345** [reply]
That worked. Thank you very much!!
